features/provider/admin/messages/inbox.feature

```diff
@@ -37,6 +37,22 @@ Feature: Audience > Messages > Inbox
       And follow "Inbox"
       Then they should see read message from "Alice" with subject "Oh, no!"
 
+    Scenario: Mark as read
+      Given a message sent from buyer "Alice" to the provider with subject "Mark me" and body "No need to read"
+      And they go to the provider inbox page
+      And they should see unread message from "Alice" with subject "Mark me"
+      When they select action "Mark as read" of "Mark me"
+      Then they should see read message from "Alice" with subject "Mark me"
```

Comment:
Do I understand correctly that already read messages cannot be marked as read. Can we have a test for it?

Reply:
Yes, read messages won't have the option to "mark as read" (in case they are marked as read anyway, rails will ignore it).

I will add a test.